public/app/features/dashboard/state/analyticsProcessor.test.ts

```diff
@@ -0,0 +1,106 @@
+import { MetaAnalyticsEventName, reportMetaAnalytics } from '@grafana/runtime';
+import { CoreApp, DataQueryRequest, DataSourceApi, dateTime, LoadingState, PanelData } from '@grafana/data';
+import { emitDataRequestEvent } from './analyticsProcessor';
+import { DashboardModel } from './DashboardModel';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const datasource = {
+  name: 'test',
+  id: 1,
+} as DataSourceApi;
+
+const dashboardModel = new DashboardModel(
+  { id: 1, title: 'Test Dashboard', uid: 'test' },
+  { folderTitle: 'Test Folder' }
+);
+
+jest.mock('app/features/dashboard/services/DashboardSrv', () => ({
+  getDashboardSrv: () => {
+    return {
+      getCurrent: () => dashboardModel,
+    };
+  },
+}));
+
+jest.mock('@grafana/runtime', () => ({
+  ...((jest.requireActual('@grafana/runtime') as unknown) as object),
```

Comment:
`...(jest.requireActual('@grafana/runtime') as any)` should work a well (untested).

Reply:
Yes, it works. Thanks!